src/pages/auth/profile/index.tsx

```diff
@@ -1,94 +1,203 @@
-import React, { useEffect } from 'react';
+import * as React from 'react';
+import { useEffect, useState } from 'react';
 import Image from 'next/image';
 import Router from 'next/router';
 import { useSession } from 'next-auth/react';
-import { resetPassword } from '../../../services';
-import { Header } from '../../../components/Global';
+import toast from 'react-hot-toast';
+import {
+  resetPassword,
+  getUserByEmail,
+  updateUsername
+} from '../../../services';
+import { Header, Input, Button } from '../../../components/Global';
+import styles from '../../../styles/pages/profile.module.scss';
 
 const ProfilePage = () => {
-  const { data: session, status } = useSession();
-  const userImage = `${session?.user?.image}#site.com/image.jpg`;
   const passwordMinLength = 8;
+  const { data: session, status } = useSession();
 
-  const [oldPass, setOldPass] = React.useState('');
-  const [password, setPassword] = React.useState('');
-  const [confirmPass, setConfirmPass] = React.useState('');
-
-  const handleChangePassword = async () => {
-    const email = session?.user?.email;
-
-    if (!email || !oldPass || !password || !confirmPass) return;
-    if (password !== confirmPass) return;
+  const [hasPassword, setHasPassword] = useState(false);
+  const [newUsername, setNewUserName] = useState('');
+  const [username, setUsername] = useState('');
+  const [userImg, setUserImg] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [oldPass, setOldPass] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPass, setConfirmPass] = useState('');
 
-    const success = await resetPassword({
-      email,
-      oldPassword: oldPass,
-      newPassword: password
-    });
-  };
+  useEffect(() => {
+    if (!session) return;
+    setUsername(session.user.name);
+    setUserImg(session.user.image);
+    setUserEmail(session.user.email);
+  }, [session]);
 
   useEffect(() => {
     if (status === 'unauthenticated') {
       Router.push('/auth/login');
     }
   }, [status]);
 
+  useEffect(() => {
+    getUserByEmail({ email: userEmail })
+      .then(user => {
+        if (user?.password) setHasPassword(true);
+        if (user.name) setUsername(user.name);
+      })
+      .catch(err => {
+        toast.error(err.message);
+      });
+  }, [userEmail]);
+
+  const handleChangePassword = async () => {
+    if (!userEmail || !oldPass || !password || !confirmPass) return;
+    if (password !== confirmPass) return;
+    try {
+      await toast.promise(
+        resetPassword({
+          email: userEmail,
+          oldPassword: oldPass,
+          newPassword: password
+        }),
+        {
+          loading: 'Changing password',
+          success: 'Password changed successfully',
+          error: err => `Error: ${err.message}`
+        }
+      );
+    } catch (err) {
+      console.error(err);
```

Comment:
Actually wait do we ever need to console errors then? I left these alone on my last review because I thought you wanted to both toast and log errors

Reply:
It's might be better to log it into a file, I'll make a Trello card

Reply:
I'll change that to toast